### PR TITLE
Ohe

### PR DESCRIPTION
**Pull request by matthewconnell**
-removed ohe function from prepR.R
-added new file for ohe

### CODE REVIEW

File: tests/test-onehot.R [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+output = onehot(fruits)
+
+test_that("Length of input and output should be equal (same number of rows)", {
+  expect_equal(length(fruits[[1]]), length(output[[1]]))
+})
+
+test_that("There should not be more than 10 unique values", {
+  expect_error(onehot(mtcars$mpg))
+})
```
Comment (jasmineqyj) [inline tests/test-onehot.R]: `mtcars$mpg` isn't a data.frame, so it shouldn't even pass the first test: `Input should be a dataframe`

File: R/onehot.R [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  # Input type check:
+  testthat::test_that("Input should be a dataframe", {
+    is.data.frame(encodable_df)
+  })
+
+  # Test that the dataframe can be reasonable encoded
+  testthat::test_that("The column should have fewer than 10 unique values", {
+    length(unique(encodable_df[[1]])) < 10
+  })
```
Comment (jasmineqyj) [inline R/onehot.R]: When I add a `print` statement after your `test_that` chunks, I can still see it printing even if I run something like `test_that("test")`. Ideally I think the function should **stop** after seeing an input error. Maybe you can try the `if...stop()` structure. [Here](https://github.ubc.ca/MDS-2019-20/DSCI_511_prog-dsci_students/blob/master/lectures/06_lecture-functions-and-testing-in-r.ipynb) are some resources.

**Review by jasmineqyj — CHANGES_REQUESTED** [review]
I tested cases when input data frame has more than 1 categorical variable, and it's currently not working.

File: tests/test-onehot.R (hunk 1 above)
Comment (matthewconnell) [reply tests/test-onehot.R]: fixed

File: R/onehot.R (hunk 2 above)
Comment (matthewconnell) [reply R/onehot.R]: Fixed and re-pushed. Thank you for the resources

**Comment by matthewconnell** [comment]
Currently does not have the functionality for dataframes with more than one categorical variable; they have to be done one column at a time. Will update documentation to reflect this and open an issue to update functionality.

**Pull request closed without merging**